### PR TITLE
Update llama-run to include temperature option

### PR DESCRIPTION
**Pull request by ericcurtin**
This commit updates the `examples/run/README.md` file to include a new option for setting the temperature and updates the `run.cpp` file to parse this option.

### CODE REVIEW

File: examples/run/run.cpp [hunk 1]
```diff
@@ -68,15 +81,24 @@ class Opt {
...
         }
 
+        ctx_params_.n_batch        = context_size_ >= 0 ? context_size_ : context_size_default;
+        model_params_.n_gpu_layers = ngl_ >= 0 ? ngl_ : ngl_default;
+        temperature_               = temperature_ >= 0 ? temperature_ : temperature_default;
+
         return 0;  // Success
     }
 
+    llama_context_params         ctx_params_;
```
Comment (ngxson) [inline examples/run/run.cpp:91]:
```suggestion
    llama_context_params         ctx_params;
```

The trailing underscore does not have clear meaning to it, I'd suggest to remove it or either add postfix, for example: `ctx_params_current`

File: examples/run/run.cpp [hunk 2]
```diff
@@ -103,6 +136,10 @@ class Opt {
                 if (handle_option_with_value(argc, argv, i, ngl_) == 1) {
                     return 1;
                 }
+            } else if (options_parsing && strcmp(argv[i], "--temperature") == 0) {
```
Comment (ngxson) [inline examples/run/run.cpp:139]: Not sure why every other places the name is `--temp` but here we got `"--temperature"`

**Review by ngxson — COMMENTED** [review]
Tbh I'm not sure what's the long-term plan for `llama-run`.

My thought is that if now we add `--temp`, I'm pretty sure someone will also add other sampling params like top-k, top-p, DRY, etc in the near future, to a point that it will defeat the initial goal of `llama-run` which is "just run".

File: examples/run/run.cpp (hunk 1 above)
Comment (ericcurtin) [reply examples/run/run.cpp]: Removed all trailing _ 's

File: examples/run/run.cpp (hunk 2 above)
Comment (ericcurtin) [reply examples/run/run.cpp]: Changed to --temp

**Comment by ericcurtin** [comment]
> Tbh I'm not sure what's the long-term plan for `llama-run`.
> 
> My thought is that if now we add `--temp`, I'm pretty sure someone will also add other sampling params like top-k, top-p, DRY, etc in the near future, to a point that it will defeat the initial goal of `llama-run` which is "just run".

I had a use case for --temp.

If somebody has a use case for extra arguments, I don't have an immediate issue with merging them. Yes, I'd hope it would be less complex than llama-cli, but personally I have no issue with people adding extra args if they need them.

**Comment by ericcurtin** [comment]
But of course, simple use cases like:

`llama-run smollm:135m`

should continue to work.

**Review by ggerganov — APPROVED** [review]

**Review by slaren — APPROVED** [review]

**Pull request merged**